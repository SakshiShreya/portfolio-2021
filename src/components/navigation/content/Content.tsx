import { useContext } from "react";
import { DeviceContext } from "../../../context/DeviceContext";
import { NavigationContext } from "../../../context/NavigationContext";
import Routes from "../../Routes";
import styles from "./Content.module.scss";

export interface ContentProps {}

const Content: React.FC<ContentProps> = () => {
  const { isOpen } = useContext(NavigationContext);
  const { device } = useContext(DeviceContext);

  return (
    <div id="content" className={[styles.cont, device === "desktop" || isOpen ? styles.slide : null].join(" ")}>
      <Routes />
    </div>
  );
};

export default Content;
