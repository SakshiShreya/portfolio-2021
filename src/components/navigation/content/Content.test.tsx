import { HashRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Content from "./Content";

it("should render content properly", () => {
  const elem = renderer
    .create(
      <HashRouter>
        <Content />
      </HashRouter>
    )
    .toJSON();
  expect(elem).toMatchSnapshot();
});
