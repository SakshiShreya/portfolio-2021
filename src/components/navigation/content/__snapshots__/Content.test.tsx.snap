// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render content properly 1`] = `
<div
  className="cont slide"
  id="content"
>
  <div
    className="cont"
  >
    <div>
      <p
        className="h1 head"
      >
        Hello World... 
      </p>
      <h2
        className="h1 head name"
      >
        I am 
        <span
          className="green"
        >
          S
        </span>
        akshi 
        <span
          className="orange"
        >
          S
        </span>
        hreya
        ,
      </h2>
      <p
        className="h1 head"
      >
        Web Developer
      </p>
      <p
        className="h2"
      >
        and Mentor.
      </p>
      <p
        className="psText"
      >
        I can develop responsive and pixel-perfect websites. I love what I do. I can mentor you to be the same.
      </p>
      <a
        className="btn secondary small border help"
        href="#/"
        onClick={[Function]}
      >
        How can I help you?
      </a>
    </div>
  </div>
</div>
`;
